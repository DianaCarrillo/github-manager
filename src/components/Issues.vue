<template>
  <div>
    <p> REPOSITORY: {{ repository.name }}</p>
    <p>Issues:</p>
    <div v-for="iss in user.issues.edges" :key="iss.id">
      <EachIssue :iss="iss" />
    </div>
    <p>Issue # </p>
    <SelectIssue :user="user" />
  </div>
</template>

<script>
import EachIssue from './EachIssue.vue'
import SelectIssue from './SelectIssue.vue'
export default {
  name: 'Issues',
  components: {
    EachIssue,
    SelectIssue,
  },
  props: {
    user: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      repository: {},
      issue: '',
    }
  },
}
</script>
