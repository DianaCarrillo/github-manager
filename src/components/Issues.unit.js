import Issues from './Issues.vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'

describe('./Issues.vue', () => {
  it('Issues is a vue instance', () => {
    let localVue = createLocalVue()
    const wrapper = shallowMount(Issues, { localVue })
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('Displayed issues correctly with query data', () => {
    let localVue = createLocalVue()
    const wrapper = shallowMount(Issues, { localVue })
    wrapper.setData({
      user: {
        issues: {
          edges: {
            node: {
              author: {
                avatarUrl:
                  'https://avatars0.githubusercontent.com/u/39560661?v=4',
                login: 'DianaCarrillo',
              },
              body:
                'Majo, recuerda que debemos hacer merge en los cambios que hicimos ayer. ¡Gracias!',
              id: 'MDU6SXNzdWUzMzYyNDIyNTQ=',
              number: 9,
              reactions: {
                edges: {
                  node: {
                    content: 'HOORAY',
                  },
                },
              },
              state: 'OPEN',
              title: 'división de archivos. Cli.js e index.js',
            },
          },
        },
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
