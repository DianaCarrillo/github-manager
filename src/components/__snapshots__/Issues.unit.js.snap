// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./Issues.vue Displayed issues correctly with query data 1`] = `
<div>
  <section
    id="select-issue"
  >
    <selectissue-stub
      user="[object Object]"
    />
  </section>
   
  <div>
    <eachissue-stub
      iss="[object Object]"
    />
  </div>
</div>
`;
